11-designing-for-mental-health/projects/final/app/src/main/java/com/raywenderlich/android/tacotuesday/tryIt/TryItRecipesFragment.kt

```diff
@@ -0,0 +1,93 @@
+ * Software in any work that is designed, intended, or marketed for pedagogical or
+ * instructional purposes related to programming, coding, application development,
+ * or information technology.  Permission for such use, copying, modification,
+ * merger, publication, distribution, sublicensing, creation of derivative works,
+ * or sale is expressly withheld.
+ *
+ * This project and source code may use libraries or frameworks that are
+ * released under various Open-Source licenses. Use of those libraries and
+ * frameworks are governed by their own individual licenses.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.raywenderlich.android.tacotuesday.tryIt
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import androidx.lifecycle.viewModelScope
+import androidx.recyclerview.widget.ItemTouchHelper
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.android.material.snackbar.Snackbar
+import com.raywenderlich.android.tacotuesday.data.Recipe
+import com.raywenderlich.android.tacotuesday.databinding.FragmentTryItRecipesListBinding
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class TryItRecipesFragment : Fragment() {
+
+  private lateinit var binding: FragmentTryItRecipesListBinding
+  private val viewModel: TryItRecipesViewModel by viewModels()
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+      savedInstanceState: Bundle?): View? {
+    binding = FragmentTryItRecipesListBinding.inflate(
+        layoutInflater, container, false)
+
+    val recipeRecyclerViewAdapter = TryItRecipesRecyclerViewAdapter(::discard, ::toggleHasMade)
+    with(binding.root) {
+      layoutManager = LinearLayoutManager(context)
+      adapter = recipeRecyclerViewAdapter
+      itemAnimator = null
+    }
+
+    ItemTouchHelper(SwipeToDiscardCallback(recipeRecyclerViewAdapter, ::discard))
+        .attachToRecyclerView(binding.root)
+
+    with(viewModel) {
+      savedRecipes.observe(viewLifecycleOwner, Observer {
+        (binding.root.adapter as TryItRecipesRecyclerViewAdapter).submitList(it)
+      })
+    }
+    return binding.root
+  }
+
+  private fun discard(recipe: Recipe) {
+    viewModel.discardRecipe(recipe, viewModel.viewModelScope)
+    Snackbar.make(binding.root,
+        "Note deleted",
```

Comment:
🚨 hard coded strings in UI, here and below 🚨 - prevents translation, which I think can be argued to be a small sliver of accessibility.

Reply:
I did a duplicate comment on this in the chapter 02. Sorry